.github/workflows/ci.yml

```diff
@@ -0,0 +1,75 @@
+name: CI/CD
+
+on:
+  pull_request:
+    types: [ opened, synchronize ]
+    branches: [ main ]
+  push:
+    branches: [ main ]
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      re_image: ${{ steps.create-inputs.outputs.re_image }}
+      re_dockerfile: ${{ steps.create-inputs.outputs.re_dockerfile }}
+      do_build: ${{ steps.create-inputs.outputs.do_build }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v40
+        with:
+          json: true
+          quotepath: false
+      - name: Create inputs
+        id: create-inputs
+        shell: python {0}
+        run: |
```

Comment:
I'm wondering if we should extract these - maybe into a `./scripts/` directory in the repo?

Reply:
Yes, I was planning to do this but ran out of time.
Could this be pushed to a future improvement?
Specifically that would be
* External scripts
* Rewrite in nodejs
* Incorporate code scanning action

Reply:
Sure. Lets get a GH issue raised to track this.